projects/distributed-tracing/src/shared/dashboard/widgets/trace-detail/data/api-trace-detail-data-source.model.ts

```diff
@@ -1,74 +1,24 @@
-import { Dictionary } from '@hypertrace/common';
-import { ARRAY_PROPERTY, Model, ModelProperty, PLAIN_OBJECT_PROPERTY } from '@hypertrace/hyperdash';
-import { EMPTY, Observable, of } from 'rxjs';
-import { mergeMap } from 'rxjs/operators';
-import { Trace, traceIdKey, traceTypeKey } from '../../../../graphql/model/schema/trace';
-import { SpecificationBuilder } from '../../../../graphql/request/builders/specification/specification-builder';
-import {
-  TraceGraphQlQueryHandlerService,
-  TRACE_GQL_REQUEST
-} from '../../../../graphql/request/handlers/traces/trace-graphql-query-handler.service';
-import { GraphQlDataSourceModel } from '../../../data/graphql/graphql-data-source.model';
+import { Model } from '@hypertrace/hyperdash';
+import { Trace } from '../../../../graphql/model/schema/trace';
+
+import { TraceDetailData, TraceDetailDataSourceModel } from './trace-detail-data-source.model';
 
 @Model({
   type: 'api-trace-detail-data-source'
 })
-export class ApiTraceDetailDataSourceModel extends GraphQlDataSourceModel<TraceDetailData> {
-  @ModelProperty({
-    key: 'trace',
-    required: true,
-    type: PLAIN_OBJECT_PROPERTY.type
-  })
-  public trace!: Trace;
-
-  @ModelProperty({
-    key: 'attributes',
-    type: ARRAY_PROPERTY.type
-  })
-  public attributes: string[] = [];
-
-  private readonly attributeSpecBuilder: SpecificationBuilder = new SpecificationBuilder();
-
-  public getData(): Observable<TraceDetailData> {
-    return this.query<TraceGraphQlQueryHandlerService>({
-      requestType: TRACE_GQL_REQUEST,
-      traceType: this.trace[traceTypeKey],
-      traceId: this.trace[traceIdKey],
-      spanLimit: 0,
-      timeRange: this.getTimeRangeOrThrow(),
-      traceProperties: this.getTraceAttributes().map(attribute =>
-        this.attributeSpecBuilder.attributeSpecificationForKey(attribute)
-      )
-    }).pipe(mergeMap(trace => this.mapResponseObject(trace)));
-  }
-
+export class ApiTraceDetailDataSourceModel extends TraceDetailDataSourceModel {
   protected getTraceAttributes(): string[] {
-    return ['tags', 'statusCode', 'traceId', 'apiTraceId', ...this.attributes];
-  }
-
-  private mapResponseObject(trace: Trace | undefined): Observable<TraceDetailData> {
-    if (trace === undefined) {
-      return EMPTY;
-    }
-
-    return of(this.constructTraceDetailData(trace));
+    return [...super.getTraceAttributes(), 'apiTraceId'];
```

Comment:
shouldn't this be fetching the `traceId` (since `apiTraceId` is already returned as the default `id` for an api trace) - this also has implications for building the detail data below, where now we have to override the id set from super (which is actually the apitraceId, but should come from traceId for an api trace) and set the entrySpanId from the default id attribute.